# <center> Assignment #1
## <center>  Exploratory data analysis with Pandas


**In this task you should use Pandas to answer a few questions about the [Adult](https://archive.ics.uci.edu/ml/datasets/Adult) dataset. **

Unique values of all features (for more information, please see the links above):
- `age`: continuous.
- `workclass`: Private, Self-emp-not-inc, Self-emp-inc, Federal-gov, Local-gov, State-gov, Without-pay, Never-worked.
- `fnlwgt`: continuous.
- `education`: Bachelors, Some-college, 11th, HS-grad, Prof-school, Assoc-acdm, Assoc-voc, 9th, 7th-8th, 12th, Masters, 1st-4th, 10th, Doctorate, 5th-6th, Preschool.
- `education-num`: continuous.
- `marital-status`: Married-civ-spouse, Divorced, Never-married, Separated, Widowed, Married-spouse-absent, Married-AF-spouse.
- `occupation`: Tech-support, Craft-repair, Other-service, Sales, Exec-managerial, Prof-specialty, Handlers-cleaners, Machine-op-inspct, Adm-clerical, Farming-fishing, Transport-moving, Priv-house-serv, Protective-serv, Armed-Forces.
- `relationship`: Wife, Own-child, Husband, Not-in-family, Other-relative, Unmarried.
- `race`: White, Asian-Pac-Islander, Amer-Indian-Eskimo, Other, Black.
- `sex`: Female, Male.
- `capital-gain`: continuous.
- `capital-loss`: continuous.
- `hours-per-week`: continuous.
- `native-country`: United-States, Cambodia, England, Puerto-Rico, Canada, Germany, Outlying-US(Guam-USVI-etc), India, Japan, Greece, South, China, Cuba, Iran, Honduras, Philippines, Italy, Poland, Jamaica, Vietnam, Mexico, Portugal, Ireland, France, Dominican-Republic, Laos, Ecuador, Taiwan, Haiti, Columbia, Hungary, Guatemala, Nicaragua, Scotland, Thailand, Yugoslavia, El-Salvador, Trinadad&Tobago, Peru, Hong, Holand-Netherlands.   
- `salary`: >50K,<=50K

In [2]:
import pandas as pd

In [3]:
data = pd.read_csv('adult.data.csv')

**1. How many men and women (*sex* feature) are represented in this dataset?** 

In [ ]:
#โด

**2. What is the average age (*age* feature) of women?**

In [1]:
#จูน

**3. What is the percentage of German citizens (*native-country* feature)?**

In [ ]:
#อาร์ม

**4-5. What are the mean and standard deviation of age for those who earn more than 50K per year (*salary* feature) and those who earn less than 50K per year?**

In [52]:
#ปอ
rich = data[data.salary == '>50K']
rich

age         workclass  fnlwgt     education  education-num  \
7       52  Self-emp-not-inc  209642       HS-grad              9   
8       31           Private   45781       Masters             14   
9       42           Private  159449     Bachelors             13   
10      37           Private  280464  Some-college             10   
11      30         State-gov  141297     Bachelors             13   
...    ...               ...     ...           ...            ...   
32539   71                 ?  287372     Doctorate             16   
32545   39         Local-gov  111499    Assoc-acdm             12   
32554   53           Private  321865       Masters             14   
32557   40           Private  154374       HS-grad              9   
32560   52      Self-emp-inc  287927       HS-grad              9   

           marital-status         occupation   relationship  \
7      Married-civ-spouse    Exec-managerial        Husband   
8           Never-married     Prof-specialty  Not-in-family   
9      Married-civ-spouse    Exec-managerial        Husband   
10     Married-civ-spouse    Exec-managerial        Husband   
11     Married-civ-spouse     Prof-specialty        Husband   
...                   ...                ...            ...   
32539  Married-civ-spouse                  ?        Husband   
32545  Married-civ-spouse       Adm-clerical           Wife   
32554  Married-civ-spouse    Exec-managerial        Husband   
32557  Married-civ-spouse  Machine-op-inspct        Husband   
32560  Married-civ-spouse    Exec-managerial           Wife   

                     race     sex  capital-gain  capital-loss  hours-per-week  \
7                   White    Male             0             0              45   
8                   White  Female         14084             0              50   
9                   White    Male          5178             0              40   
10                  Black    Male             0             0              80   
11     Asian-Pac-Islander    Male             0             0              40   
...                   ...     ...           ...           ...             ...   
32539               White    Male             0             0              10   
32545               White  Female             0             0              20   
32554               White    Male             0             0              40   
32557               White    Male             0             0              40   
32560               White  Female         15024             0              40   

      native-country salary  
7      United-States   >50K  
8      United-States   >50K  
9      United-States   >50K  
10     United-States   >50K  
11             India   >50K  
...              ...    ...  
32539  United-States   >50K  
32545  United-States   >50K  
32554  United-States   >50K  
32557  United-States   >50K  
32560  United-States   >50K  

[7841 rows x 15 columns]

In [32]:
rich_mean = rich['age'].mean()
rich_mean

44.24984058155847

จะเห็นได้ว่าคนรายได้สูงมีอายุเฉลี่ยอยู่ที่ **44.24984058155847**

In [33]:
rich_std = rich['age'].std()
rich_std

10.519027719851826

จะเห็นได้ว่าคนรายได้สูงมีค่า Standard Deviation อยู่ที่ **10.5190277198851826**

In [ ]:
#โด

**6. Is it true that people who earn more than 50K have at least high school education? (*education – Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters* or *Doctorate* feature)**

In [ ]:
#จูน

**7. Display age statistics for each race (*race* feature) and each gender (*sex* feature). Use *groupby()* and *describe()*. Find the maximum age of men of *Amer-Indian-Eskimo* race.**

In [ ]:
#อาร์ม

**8. Among whom is the proportion of those who earn a lot (>50K) greater: married or single men (*marital-status* feature)? Consider as married those who have a *marital-status* starting with *Married* (Married-civ-spouse, Married-spouse-absent or Married-AF-spouse), the rest are considered bachelors.**

In [21]:
#ปอ
married = data.loc[(data['sex'] == 'Male') & (data['marital-status'].str.startswith('Married')), 'salary'
        ].value_counts()
married

<=50K    7576
>50K     5965
Name: salary, dtype: int64

In [22]:
single_men = data.loc[(data['sex'] == 'Male') & (~data['marital-status'].str.startswith('Married')), 'salary'].value_counts()
single_men

<=50K    7552
>50K      697
Name: salary, dtype: int64

จะเห็นได้ว่าคนที่แต่งงานแล้วมีรายได้มากกว่า 50K จะอยู่ที่ 5,965 คน ส่วนคนที่ยังโสดอยู่มีเพียง 697 คน 
**จึงสรุปได้ว่าคนที่แต่งงานแล้วมีคนมี่รายได้เยอะมากกว่า**

**9. What is the maximum number of hours a person works per week (*hours-per-week* feature)? How many people work such a number of hours, and what is the percentage of those who earn a lot (>50K) among them?**

In [4]:
#คู่ จูน โดนัท

In [ ]:
# You code here

**10. Count the average time of work (*hours-per-week*) for those who earn a little and a lot (*salary*) for each country (*native-country*). What will these be for Japan?**

In [34]:
data

age         workclass  fnlwgt   education  education-num  \
0       39         State-gov   77516   Bachelors             13   
1       50  Self-emp-not-inc   83311   Bachelors             13   
2       38           Private  215646     HS-grad              9   
3       53           Private  234721        11th              7   
4       28           Private  338409   Bachelors             13   
...    ...               ...     ...         ...            ...   
32556   27           Private  257302  Assoc-acdm             12   
32557   40           Private  154374     HS-grad              9   
32558   58           Private  151910     HS-grad              9   
32559   22           Private  201490     HS-grad              9   
32560   52      Self-emp-inc  287927     HS-grad              9   

           marital-status         occupation   relationship   race     sex  \
0           Never-married       Adm-clerical  Not-in-family  White    Male   
1      Married-civ-spouse    Exec-managerial        Husband  White    Male   
2                Divorced  Handlers-cleaners  Not-in-family  White    Male   
3      Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4      Married-civ-spouse     Prof-specialty           Wife  Black  Female   
...                   ...                ...            ...    ...     ...   
32556  Married-civ-spouse       Tech-support           Wife  White  Female   
32557  Married-civ-spouse  Machine-op-inspct        Husband  White    Male   
32558             Widowed       Adm-clerical      Unmarried  White  Female   
32559       Never-married       Adm-clerical      Own-child  White    Male   
32560  Married-civ-spouse    Exec-managerial           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country salary  
0              2174             0              40  United-States  <=50K  
1                 0             0              13  United-States  <=50K  
2                 0             0              40  United-States  <=50K  
3                 0             0              40  United-States  <=50K  
4                 0             0              40           Cuba  <=50K  
...             ...           ...             ...            ...    ...  
32556             0             0              38  United-States  <=50K  
32557             0             0              40  United-States   >50K  
32558             0             0              40  United-States  <=50K  
32559             0             0              20  United-States  <=50K  
32560         15024             0              40  United-States   >50K  

[32561 rows x 15 columns]

In [41]:
#คู่ อาร์ม ปอ
rich = data[data.salary == '>50K']
rich

age         workclass  fnlwgt     education  education-num  \
7       52  Self-emp-not-inc  209642       HS-grad              9   
8       31           Private   45781       Masters             14   
9       42           Private  159449     Bachelors             13   
10      37           Private  280464  Some-college             10   
11      30         State-gov  141297     Bachelors             13   
...    ...               ...     ...           ...            ...   
32539   71                 ?  287372     Doctorate             16   
32545   39         Local-gov  111499    Assoc-acdm             12   
32554   53           Private  321865       Masters             14   
32557   40           Private  154374       HS-grad              9   
32560   52      Self-emp-inc  287927       HS-grad              9   

           marital-status         occupation   relationship  \
7      Married-civ-spouse    Exec-managerial        Husband   
8           Never-married     Prof-specialty  Not-in-family   
9      Married-civ-spouse    Exec-managerial        Husband   
10     Married-civ-spouse    Exec-managerial        Husband   
11     Married-civ-spouse     Prof-specialty        Husband   
...                   ...                ...            ...   
32539  Married-civ-spouse                  ?        Husband   
32545  Married-civ-spouse       Adm-clerical           Wife   
32554  Married-civ-spouse    Exec-managerial        Husband   
32557  Married-civ-spouse  Machine-op-inspct        Husband   
32560  Married-civ-spouse    Exec-managerial           Wife   

                     race     sex  capital-gain  capital-loss  hours-per-week  \
7                   White    Male             0             0              45   
8                   White  Female         14084             0              50   
9                   White    Male          5178             0              40   
10                  Black    Male             0             0              80   
11     Asian-Pac-Islander    Male             0             0              40   
...                   ...     ...           ...           ...             ...   
32539               White    Male             0             0              10   
32545               White  Female             0             0              20   
32554               White    Male             0             0              40   
32557               White    Male             0             0              40   
32560               White  Female         15024             0              40   

      native-country salary  
7      United-States   >50K  
8      United-States   >50K  
9      United-States   >50K  
10     United-States   >50K  
11             India   >50K  
...              ...    ...  
32539  United-States   >50K  
32545  United-States   >50K  
32554  United-States   >50K  
32557  United-States   >50K  
32560  United-States   >50K  

[7841 rows x 15 columns]

In [48]:
average_rich = rich.groupby('native-country').mean()
average_rich['hours-per-week']

native-country
?                     45.547945
Cambodia              40.000000
Canada                45.641026
China                 38.900000
Columbia              50.000000
Cuba                  42.440000
Dominican-Republic    47.000000
Ecuador               48.750000
El-Salvador           45.000000
England               44.533333
France                50.750000
Germany               44.977273
Greece                50.625000
Guatemala             36.666667
Haiti                 42.750000
Honduras              60.000000
Hong                  45.000000
Hungary               50.000000
India                 46.475000
Iran                  47.500000
Ireland               48.000000
Italy                 45.400000
Jamaica               41.100000
Japan                 47.958333
Laos                  40.000000
Mexico                46.575758
Nicaragua             37.500000
Peru                  40.000000
Philippines           43.032787
Poland                39.000000
Portugal              41.

In [50]:
poor = data[data.salary == '<=50K']
average_poor = poor.groupby('native-country').mean()
average_poor['hours-per-week']

native-country
?                             40.164760
Cambodia                      41.416667
Canada                        37.914634
China                         37.381818
Columbia                      38.684211
Cuba                          37.985714
Dominican-Republic            42.338235
Ecuador                       38.041667
El-Salvador                   36.030928
England                       40.483333
France                        41.058824
Germany                       39.139785
Greece                        41.809524
Guatemala                     39.360656
Haiti                         36.325000
Holand-Netherlands            40.000000
Honduras                      34.333333
Hong                          39.142857
Hungary                       31.300000
India                         38.233333
Iran                          41.440000
Ireland                       40.947368
Italy                         39.625000
Jamaica                       38.239437
Japan                    

**จะเห็นได้ว่าคนที่มีรายได้สูงจะมีชั่วโมงในการทำงานที่สูงกว่า**